frontend/ui/widget/radiomark.lua

```diff
@@ -11,16 +13,25 @@ local RadioMark = InputContainer:new{
     face = Font:getFace("smallinfofont"),
     baseline = 0,
     _mirroredUI = BD.mirroredUILayout(),
+    -- round radio mark looks a little bit upper than the button/menu text
+    -- default vertical down shift ratio (to the text height) looks good in touchmenu
+    v_shift_ratio = 0.05,
 }
 
 function RadioMark:init()
-    local widget = TextWidget:new{
+    local text_widget = TextWidget:new{
         text = self.checkable and (self.checked and "◉ " or "◯ ") or "",
         face = self.face,
         fgcolor = self.enabled and Blitbuffer.COLOR_BLACK or Blitbuffer.COLOR_DARK_GRAY,
         para_direction_rtl = self._mirroredUI,
     }
-    self.baseline = widget:getBaseline()
+    self.baseline = text_widget:getBaseline()
+    local pad = math.floor(text_widget:getSize().h * self.v_shift_ratio)
```

Comment:
Can you try just setting  `text_widget.forced_baseline` after the creation, as it's only used at :paintTo() time:
https://github.com/koreader/koreader/blob/cb7e670c4a73a21db0db829f67b8014658ce4bb2/frontend/ui/widget/textwidget.lua#L357

```lua
self.baseline = text_widget:getBaseline()
self.baseline = self.baseline + math.floor(text_widget:getSize().h * self.v_shift_ratio)
text_widget.forced_baseline = self.baseline
```

No VerticalGroupVerticalSpan would be needed (they would change the height of the widget, dunno which side effects this may have).